arch/riscv/kernel/head.S

```diff
@@ -198,6 +199,17 @@ ENTRY(_start_kernel)
 	csrw CSR_IE, zero
 	csrw CSR_IP, zero
 
+#ifdef CONFIG_RISCV_COVE_GUEST_PROMOTE
+	mv s1, a0
+	mv s2, a1
+	li a7, COVE_PROMOTE_SBI_EXT_ID
+	li a6, COVE_PROMOTE_SBI_FID
+	mv a0, a1
```

Comment:
Regarding the updated patch, we can't invoke SBI calls like this without checks. It will just fail if the higher privilege mode system doesn't support it.

Reply:
This piece of code is active only for non-M mode system (HS/VS mode) compiled with dedicated kernel build flag `CONFIG_RISCV_COVE_GUEST_PROMOTE`, so won't execute for a M-mode kernel or a regular COVE guest created in a multi-step fashion. If the more privileged mode doesn't support `promote_to_tvm` call, then it will return an SBI error reflected in a0/a1 gprs. We can ignore the SBI result here the same as we would do it at any later guest's execution stage. This is because the result of promotion is reflected via attestation. The VM that failed the promotion will continue as a regular VM and won't retrieve secrets from attestation.